proto/frequenz/api/reporting/v1/reporting.proto

```diff
@@ -0,0 +1,534 @@
+// formula-driven historical data aggregation. For detailed information on component
+// categories and the metrics they provide, please consult the referenced component categories
+// imported above.
+//
+service Reporting {
+  // Fetches historical metrics for a list of microgrid components.
+  //
+  // !!! note
+  //     This RPC endpoint retrieves historical metrics data for components of one or more
+  //     microgrids. Clients need to provide at least one microgrid ID and one component ID
+  //     to get a result.
+  rpc ListMicrogridComponentsData(ListMicrogridComponentsDataRequest)
+    returns (ListMicrogridComponentsDataResponse) {}
+
+  // Streams metrics for a list of microgrid components.
+  //
+  // !!! note
+  //     This RPC endpoint streams metrics data for components of one or more microgrids.
+  //     Clients need to provide at least one component ID to get a result.
+  rpc ReceiveMicrogridComponentsDataStream(ReceiveMicrogridComponentsDataStreamRequest)
+    returns (stream ReceiveMicrogridComponentsDataStreamResponse) {}
+
+  // Fetches aggregated historical metrics based on user-defined formulas for a list of
+  // microgrid components.
+  //
+  // !!! note
+  //     This RPC endpoint retrieves aggregated historical metrics data based on user-defined
+  //     formulas for microgrid components. At least one formula must be provided for a valid query.
+  rpc ListAggregatedMicrogridComponentsData(ListAggregatedMicrogridComponentsDataRequest)
+    returns (ListAggregatedMicrogridComponentsDataResponse) {}
+
+  // Streams aggregated metrics based on user-defined formulas for a list of microgrid
+  // components.
+  //
+  // !!! note
+  //     This RPC endpoint retrieves aggregated historical metrics data based on user-defined
+  //     formulas for microgrid components. At least one formula for one microgrid must be provided
+  //     for a valid result.
+  rpc ReceiveAggregatedMicrogridComponentsDataStream(ReceiveAggregatedMicrogridComponentsDataStreamRequest)
+    returns (ReceiveAggregatedMicrogridComponentsDataStreamResponse) {}
+}
+
+// Time-based filter for querying aggregated microgrid components data.
+//
+// !!! note
+//     Specifies the start and end time for the query. Both fields are optional, but
+//     their absence or presence has specific implications for the query results.
+//
+message TimeFilter {
+  // Optional UTC start time for the query.
+  //
+  // !!! info
+  //     If not provided, the query defaults to the earliest available data for the
+  //     specified microgrid components.
+  google.protobuf.Timestamp start = 1;
+
+  // Optional UTC end time for the query.
+  //
+  // !!! info
+  //     If not provided, the query defaults to the most recent available data for the
+  //     specified microgrid components.
+  google.protobuf.Timestamp end = 2;
+}
+
+// Resampling options for aggregated microgrid components data.
+message ResamplingOptions {
```

Comment:
We need to clarify how resampling is performed. Is the resulting timestamp the oldest, newest or some other timestamp of the time intervall? Also state which aggregation is performed, which will be average by default. 

In future we can add further aggregations. 

For reference, I suggest to closely follow pandas' resampling https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.resample.html.

Reply:
This feels like a broad topic. I'll suggest creating an issue for this, since we will very likely not touch aggregations in phase 1.

Reply:
```
// Resampling options for aggregated microgrid components data.
// If data is resampled, all samples that fall in a left-closed time interval 
// determined by the resolution will be aggregated. 
// At the moment only mean aggregation is supported.
// The timestamp of the aggregated sample corresponds to the oldest 
// possible timestamp of the time interval. 
```
Whether we default to oldest or newest is just a matter convention though.